### PR TITLE
chore:statistic canteen views

### PR DESCRIPTION
**Pull request by CommanderStorm**
This PR migrates all parts which were canteen subsystem which were manually tallying stuff to db-views

This is a draft until the canteen test you have teasered are done ^^

### CODE REVIEW

File: server/backend/migration/20231015000000.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+// migrates the static data for the canteen rating system and adds the necessary cronjob entries
+func (m TumDBMigrator) migrate20231015000000() *gormigrate.Migration {
+	return &gormigrate.Migration{
+		ID: "20231015000000",
+		Migrate: func(tx *gorm.DB) error {
+			// cronjob
+			if err := tx.Delete(&model.Crontab{}, "type = 'averageRatingComputation'").Error; err != nil {
+				return err
+			}
+			if err := SafeEnumAdd(tx, &model.Crontab{}, "type", "averageRatingComputation"); err != nil {
```
Comment (CommanderStorm) [inline server/backend/migration/20231015000000.go:100]:
```suggestion
			if err := SafeEnumRemove(tx, &model.Crontab{}, "type", "averageRatingComputation"); err != nil {
```

File: server/backend/migration/20231015000000.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+				if err := tx.Exec("DROP VIEW IF EXISTS " + view).Error; err != nil {
+					return err
+				}
+			}
+			// tables
+			if err := tx.AutoMigrate(&CafeteriaRatingAverage{}, &DishRatingAverage{}, &DishRatingTagAverage{}, &CafeteriaRatingTagsAverage{}, &DishNameTagAverage{}); err != nil {
+				return err
+			}
+			// cronjob
+			if err := SafeEnumRemove(tx, &model.Crontab{}, "type", "averageRatingComputation"); err != nil {
```
Comment (CommanderStorm) [inline server/backend/migration/20231015000000.go:158]:
```suggestion
			if err := SafeEnumAdd(tx, &model.Crontab{}, "type", "averageRatingComputation"); err != nil {
```

**Comment by tobiasjungmann** [comment]
The cafeteria tests are following next week. Hopefully they meet the expectations after the teaser :D

**Comment by CommanderStorm** [comment]
@tobiasjungmann ping

**Comment by CommanderStorm** [comment]
merging as this makes my live significantly eazier

**Pull request merged**